### DISCUSSION TITLE
next/future/image sizes warning

### DISCUSSION THREAD

**Discussion by ahetawal-p** [vercel/next.js#40643] [Help]
Hi,
I am migrating our current image component to use the the newly release 12.3 stable next/future/image component.
My new component looks like this:

```
 image: {
    objectFit: 'cover',
  },

<Image
        src={getImageUrl(url)}
        fill
        sizes="100vw"
        alt=""
        className={classes.image}
      />
```
With this setup, I am still seeing this warning showing up in my console.
`Image with src "xxxx" has "fill" but is missing "sizes" prop. Please add it to improve page performance. Read more: https://nextjs.org/docs/api-reference/next/future/image#sizes`

How can I get rid of this warning, it seems to be having the correct attributes as such ?

**Comment by chrisweb** [comment]
@ahetawal-p the code considers a sizes value of "100vw" to be equivalent to no sizes:

```js
!img.getAttribute('sizes') || img.getAttribute('sizes') === '100vw'
```

there are several other conditions too, but this one is the reason for the warning

now I have a case where I want to display a hero image that is as wide as the page and do this for every resolution, my code looks like this:

```js
<Image
    src="/images/foo.png"
    alt="bar"
    fill
    style={{ objectFit: 'cover' }}
    sizes="100vw"
    priority
    quality={80}
/>
```

I think this should be good, but next.js (I currently use next.js v13.1.6) tells me I need to specify sizes, I think this is a false positive but I'm not totally sure

> Image with src "/images/foo.png" has "fill" but is missing "sizes" prop. Please add it to improve page performance.

as I understand it next/image will generate several versions with different sizes and the browser will always take the one that matches the window width or is larger than the window width

I don't want to manually add sizes as I'm afraid I might make it worse, based on my understanding I currently think the default behavior is the best choice

so not sure this is something that should get fixed or if I do something wrong?

**Comment by mrigo** [comment]
Ehy, any news about this bug?

Thanks
\m

**Reply by ollebergkvist** [reply]
Would also love to see this fixed

**Comment by ical10** [comment]
Would like to know how to remove this warning as well. 

Currently adding `sizes="100vw"` won't work, I need to specify a value (e.g. `sizes="100px"`) to remove this warning.

**Comment by youjin-10** [comment]
having the same issue - someone please help!

**Comment by xBurnsed** [comment]
Same issue here!

**Comment by AndriiSuch** [comment]
Just added the sizes="100%" to my images and it helped 
```
<Image
              className="duration-200 hover:scale-110"
              sizes="100%"
              fill
            />
```

**Reply by yutung-cheng** [reply]
This solution works for me.

**Reply by yutung-cheng** [reply]
This solution works for me again....

**Reply by cibulka** [reply]
Unfortunately `sizes="100%"` is not valid, see https://github.com/vercel/next.js/discussions/40643#discussioncomment-7598340.

**Comment by HYEOK9** [comment]
sizes="auto" also worked for me

**Reply by cibulka** [reply]
Unfortunately `sizes="auto"` is not valid, see https://github.com/vercel/next.js/discussions/40643#discussioncomment-7598340.

**Comment by cibulka** [comment]
According to developer.mozilla.org, [none of the proposed solutions are valid usage of HTML sizes](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes).

> The source size value is a [CSS length](https://developer.mozilla.org/en-US/docs/Web/CSS/length). It may be specified using font-relative units (such as em or ex), absolute units (such as px or cm), or the vw unit, which lets you specify the width as a percentage of the viewport width (1vw being 1% of the viewport width).

I understand this the way that we can use only:
* font-relative units (em, rem, ex, ...)
* absolute units (px, cm, pt, ...)
* viewport width unit (vw)

... for this reason the only way how to mark 100% viewport size is `100vw`.

**Reply by fresonn** [reply]
Yes related to **MDN**

_For the slot width, rather than providing an absolute width (for example, 480px), you can alternatively provide a width relative to the viewport (for example, 50vw) — **but not a percentage**._

**Reply by cibulka** [reply]
I've opened an Issue on Next.js repo: https://github.com/vercel/next.js/issues/58586

**Reply by fresonn** [reply]
> I've opened an Issue on Next.js repo: https://github.com/vercel/next.js/issues/58586

Good!

**Comment by ryuji-orca** [comment]
This was the solution for me.
`<Image　src={src}　alt={alt}  fill  sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw" />`

**Comment by shivaramr** [comment]
Hi Team, I tried sizes="auto", works for me

              <Image src={imgPath} alt="portfolio" fill sizes="auto" className="object-contain" />

Please check and let know.
